pkg/hubble/monitor/consumer.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/pkg/hubble/observer"
+	observerTypes "github.com/cilium/cilium/pkg/hubble/observer/types"
+	"github.com/cilium/cilium/pkg/monitor/agent/consumer"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
+)
+
+type hubbleConsumer struct {
+	observer      observer.GRPCServer
+	numEventsLost uint64
+}
+
+// NewHubbleConsumer returns an initialized pointer to hubbleConsumer.
+func NewHubbleConsumer(observer observer.GRPCServer) consumer.MonitorConsumer {
```

Comment:
nit: since this is in a sub-package of `hubble`, why not just call it `consumer`/`NewConsumer`? I think that `monitor.NewConsumer` looks better also because it it's import-renamed it'll be something like `hubbleMonitor.NewConsumer` rather than `hubbleMonitor.NewHubbleConsumer`.